### ISSUE TITLE
problem when set fix_gamma=True in batchnorm

### ISSUE THREAD

**Issue by solin319** [apache/incubator-mxnet#9624]
If fix_gamma is true, then set gamma to 1 and its gradient to 0.
But the value of gamma will be changed during parameters update. So the gamma saved in param file was not 1. These will bring a problem in convert MXNet parameters to other deep-learning platforms.
This problem was caused by we set a default weight-decay in SGD optimizer.
We must define variable gamma with wd_mult=0 to fix gamma=1 during training.

Can MXNet set wd of gamma to 0 automatically when fix_gamma=1?

**Comment by piiswrong** [comment]
This is a legacy design defect. When fix_gamma is true there shouldn't be a gamma parameter.
In the future this could be solved by creating new operator batch_norm and deprecating BatchNorm

**Comment by spidyDev** [comment]
@sandeep-krishnamurthy : Tag: Bug

**Comment by anushthakalia** [comment]
@solin319 how did you check the value of gamma? I suspect I am facing the same issue and hence, want to verify. But unfortunately, I couldn't find gamma in either arg_params or aux_params.

**Comment by vandanavk** [comment]
@mxnet-label-bot add [Operator]

**Comment by Vikas89** [comment]
@solin319 I guess, the fix is already merged and closed. 
@anirudh2290 Can we close this one.

**Comment by anirudh2290** [comment]
@Vikas89 the fix is only for coreml converter. The operator hasnt been fixed yet.

**Comment by kohillyang** [comment]
@solin319 Does it means that the parameters would be updated even its grad_req is set to "null" if the wd_mult is not set to zero?

I think this behavior is unexpected.

**Comment by szha** [comment]
I think @wkcn fixed this issue in https://github.com/apache/incubator-mxnet/pull/18500
